### ISSUE TITLE
IPython directive should strip ansi escape sequence from output.

### ISSUE THREAD

**Issue by Carreau** [ipython/ipython#11424]
Stderr/out and especially traceback get garbled in the out with Ansi escape sequence. 
We should figure out how to turn coloring off, or strip escape sequences.

**Comment by mrecachinas** [comment]
@Carreau Trying to reproduce this. Can you provide an example? I'm on macOS 10.14 using Python 3.6 and a fresh pull of ipython (running in a virtualenv).

```
$ bin/ipython
Python 3.6.5 (default, Apr 25 2018, 14:23:58)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.1.0.dev -- An enhanced Interactive Python. Type '?' for help.

In [1]: raise
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-1-9c9a2cba73bf> in <module>
----> 1 raise

RuntimeError: No active exception to reraise

In [2]: raise ValueError('\v')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-47eaec54718e> in <module>
----> 1 raise ValueError('\v')

ValueError:

In [3]:
```

**Comment by Carreau** [comment]
You can see it just **above** [this header](https://ipython.readthedocs.io/en/latest/development/ipython_directive.html#ipython-sphinx-directive-module), note that it only appears on the _documentation_ page, not in IPython shell itself:

<img width="731" alt="screen shot 2018-10-24 at 11 18 44 am" src="https://user-images.githubusercontent.com/335567/47452275-ad692380-d77e-11e8-9b08-1c119c52e9fa.png">

**Comment by ankitkariryaa** [comment]
> You can see it just **above** [this header](https://ipython.readthedocs.io/en/latest/development/ipython_directive.html#ipython-sphinx-directive-module), note that it only appears on the _documentation_ page, not in IPython shell itself:
> 
> <img alt="screen shot 2018-10-24 at 11 18 44 am" width="731" src="https://user-images.githubusercontent.com/335567/47452275-ad692380-d77e-11e8-9b08-1c119c52e9fa.png">

Hey,

the link is doesn't exist. Can you please update it?

**Comment by mrecachinas** [comment]
@ankitkariryaa https://ipython.readthedocs.io/en/stable/sphinxext.html

**Comment by mrecachinas** [comment]
@Carreau I looked at this briefly, but when I tried running the manual test, it produced no output.

**Comment by mrecachinas** [comment]
@Carreau I couldn't replicate this issue in Python 3 using Sphinx==1.8.4 on either a clone of IPython from October or today. My steps were:

```bash
git clone https://github.com/ipython/ipython
cd ipython
virtualenv -p `which python3` .
source bin/activate
pip install numpy matplotlib
python setup.py install
cd docs
make
```

This is the resulting HTML file showing the correct syntax highlighting (and no ansi escape characters).
<img width="1392" alt="screen shot 2019-02-25 at 10 59 58 am" src="https://user-images.githubusercontent.com/924243/53329850-fa076b80-38e4-11e9-8b73-5e8063d025e6.png">

I also tried the sphinx extension out on an example project using the following

```
pip install Sphinx
sphinx-quickstart
[...]
make html
```

I accepted most defaults in the `conf.py`, with the only difference in `extensions`, shown below
```
extensions = [
    'sphinx.ext.doctest',
    'sphinx.ext.intersphinx',
    'sphinx.ext.todo',
    'sphinx.ext.coverage',
    'sphinx.ext.mathjax',
    'sphinx.ext.ifconfig',
    'sphinx.ext.viewcode',
    'sphinx.ext.githubpages',
    'IPython.sphinxext.ipython_console_highlighting',
    'IPython.sphinxext.ipython_directive'
]
```

This resulted in
<img width="1392" alt="screen shot 2019-02-25 at 11 00 22 am" src="https://user-images.githubusercontent.com/924243/53329917-2fac5480-38e5-11e9-9213-d87fa3324adb.png">

Unsure why the error still shows on https://ipython.readthedocs.io/en/stable/sphinxext.html, but it doesn't appear to be reproducible. Can we trigger a rebuild of the readthedocs.io build?

**Comment by Cheburusska** [comment]
@mrecachinas you can easily replicate this issue in datatable package:

```
.. ipython:: python

    import datatable as dt
    DT = dt.Frame(A=range(5))
    DT
```

generates

![image](https://user-images.githubusercontent.com/19714042/54186268-12b67a80-44bc-11e9-8d15-935693a9dc35.png)


whilie in Jupyter notebooks it looks well:

![image](https://user-images.githubusercontent.com/19714042/54186329-37125700-44bc-11e9-990e-9415021b9505.png)

There is a colorcoding for different data types

**Comment by Cheburusska** [comment]
@Carreau maybe you could suggest where to start to investigate this issue?

**Comment by Carreau** [comment]
Hum, 

Thanks both for trying. I think so far we need to manage to narrow down where the big might be, or at least which extension/plugin/version of what is the culprit.

**Comment by Cheburusska** [comment]
@Carreau maybe there is a workaround? Can this color coding be turned off? or maybe using jupyter notebook of older version?

**Comment by Carreau** [comment]
I don't believe it can be turned off. It is most likely completely unrelated to the version of the notebook package.
